Lib/unittest/suite.py

```diff
@@ -304,11 +311,11 @@ def _tearDownPreviousClass(self, test, result):
             if tearDownClass is not None:
                 try:
                     tearDownClass()
-                except Exception as e:
+                except Exception:
                     if isinstance(result, _DebugResult):
                         raise
                     className = util.strclass(previousClass)
-                    self._createClassOrModuleLevelException(result, e,
+                    self._createClassOrModuleLevelException(result, sys.exc_info(),
```

Comment:
I think it would be better to not change this, and turn `exc` into `class, exc, traceback` at the site that needs it.

Reply:
Reverted that, and did break a ref cycle manually when "--debug".
(Alternatively it would be possible to remove the `e`' s and the `exc` parameter completely and just use `sys.exc_info()` / `info=None`  at using site. `case._addError`,  `result.addFailure`, `result.addSubTest`, ... all have the info tuple parameter only.)